src/irr/asset/CGLTFLoader.cpp

```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2020 AnastaZIuk
+// This file is part of the "Irrlicht Engine".
+// For conditions of distribution and use, see copyright notice in irrlicht.h
+
+#include "CGLTFLoader.h"
+
+#ifdef _IRR_COMPILE_WITH_GLTF_LOADER_
+
+#include "simdjson/singleheader/simdjson.h"
+#include "os.h"
+
+namespace irr
+{
+	namespace asset
+	{
+		bool CGLTFLoader::isALoadableFileFormat(io::IReadFile* _file) const
+		{
+			// TODO: change it when implementing .glb
+			return true;
+		}
+
+		asset::SAssetBundle CGLTFLoader::loadAsset(io::IReadFile* _file, const asset::IAssetLoader::SAssetLoadParams& _params, asset::IAssetLoader::IAssetLoaderOverride* _override, uint32_t _hierarchyLevel)
+		{
+			simdjson::dom::parser parser;
+			simdjson::dom::element tweets = parser.load(_file->getFileName().c_str());
```

Comment:
is there a way to have the parser load from memory, or from a provided callback?

Reply:
I can't see such option

Reply:
we need that option, the engine must always be able to load from Virtual Files (using `file->read`) because sometimes the filename will be invalid (not be a path to a legit existing file on the filesystem)

Otherwise you wont be able to load glTF from zip or other archives, etc.

Reply:
I will do the research first in the implementation, maybe it is possible

Reply:
`parser.parse`